src/negotiation.rs

```diff
@@ -20,6 +20,7 @@
 //! - A peer address, used to connect to the other peer
 
 use bitcoin::secp256k1::PublicKey;
+use chrono::{NaiveDateTime, Utc};
```

Comment:
What functions are you importing here that are not available in `std::time::SystemTime`?